rtgui/cursormanager.cc

```diff
@@ -36,98 +36,133 @@ void CursorManager::init (Glib::RefPtr<Gdk::Window> mainWindow)
-        window->set_cursor (cLeftTanMove);
-    } else if (shape == CSMoveRight) {
-        window->set_cursor (cRightTanMove);
-    } else if (shape == CSStraighten) {
-        window->set_cursor (cHand);
-    } else if (shape == CSWait) {
-        window->set_cursor (cWait);
-    } else if (shape == CSPlus) {
-        window->set_cursor (cAdd);
-    } else if (shape == CSEmpty) {
-        window->set_cursor (cHidden);
+        case CursorShape::CSAddColPicker:
+            window->set_cursor(cAddPicker);
+            break;
+        case CursorShape::CSArrow:
+            window->set_cursor(); // set_cursor without any arguments to select system default
+            break;
+        case CursorShape::CSCropSelect:
+            window->set_cursor(cCropDraw);
+            break;
+        case CursorShape::CSCrosshair:
+            window->set_cursor(cCrosshair);
+            break;
+        case CursorShape::CSEmpty:
+            window->set_cursor(cEmpty);
+            break;
+        case CursorShape::CSHandClosed:
+            window->set_cursor(cHandClosed);
+            break;
+        case CursorShape::CSHandOpen:
+            window->set_cursor(cHandOpen);
+            break;
+        case CursorShape::CSMove:
+            window->set_cursor(cHandClosed);
+            break;
+        case CursorShape::CSMove1DH:
+            window->set_cursor(cMoveX);
+            break;
+        case CursorShape::CSMove1DV:
+            window->set_cursor(cMoveY);
+            break;
+        case CursorShape::CSMove2D:
+            window->set_cursor(cMoveXY);
+            break;
+        case CursorShape::CSMoveLeft:
+            window->set_cursor(cMoveL);
+            break;
+        case CursorShape::CSMoveRight:
+            window->set_cursor(cMoveR);
+            break;
+        case CursorShape::CSMoveRotate:
+            window->set_cursor(cRotate);
+            break;
+        case CursorShape::CSPlus:
+            window->set_cursor(cAdd);
+            break;
+        case CursorShape::CSResizeBottomLeft:
+            window->set_cursor(cMoveBL);
+            break;
+        case CursorShape::CSResizeBottomRight:
+            window->set_cursor(cMoveBR);
+            break;
+        case CursorShape::CSResizeDiagonal:
+            window->set_cursor(cMoveXY);
+            break;
+        case CursorShape::CSResizeHeight:
+            window->set_cursor(cMoveY);
+            break;
+        case CursorShape::CSResizeTopLeft:
+            window->set_cursor(cMoveTL);
+            break;
+        case CursorShape::CSResizeTopRight:
+            window->set_cursor(cMoveTR);
+            break;
+        case CursorShape::CSResizeWidth:
+            window->set_cursor(cMoveX);
+            break;
+        case CursorShape::CSSpotWB:
+            window->set_cursor(cWB);
+            break;
+        case CursorShape::CSStraighten:
+            window->set_cursor(cRotate);
+            break;
+        case CursorShape::CSUndefined:
```

Comment:
@Floessie is it good practice to include this `CSUndefined` item in the switch?

Reply:
This `switch` has a `default`. So, if you want `CSUndefined` to do something different than the default, you must include it.

The thing with `switch` is: If you want to catch unhandled `enum` values you must omit the `default`. Only then will the compiler warn you if you've added an `enum` value and forgot to handle it in the `switch`. Here's an example:

```c++
enum class MyEnum {
    ITEM_A,
    ITEM_B
};

void myFunction(int my_enum_value) // Notice the `int`
{
    switch (MyEnum(my_enum_value)) { // This is mandatory for `enum class` but not for `enum`. In any case, the compiler needs to know `my_enum_value` isn't a plain `int`.
        case MyEnum::ITEM_A: {
            // Do what must be done
            return;
        }

        case MyEnum::ITEM_B: {
            // Do what must be done
            return;
        }
    }
    
    // Unhandled: Default code here
}

void myOtherFunction(MyEnum my_enum_value)
{
    bool handled = false;
    
    switch (my_enum_value) {
        case MyEnum::ITEM_A: {
            // Do what must be done
            handled = true
            break;
        }

        case MyEnum::ITEM_B: {
            // Do what must be done
            handled = true
            break;
        }
    }

    if (!handled) {
        // Unhandled: Default code here
    }
    
    // Other code here
}
```

HTH,
Flössie

Reply:
Thank you @Floessie 
I tried that, but a grep returned a whole lot of places where this enum is used, and as expected it caused compilation failures which I tried but am not skilled enough to fix. Would it be ok to commit the code as-is?

Reply:
@Beep6581 Of course! You asked about the need for including `CSUndefined` and I just noted that it must be included in this switch, because it has a different `default`. No need to change anything. The example given above just outlines good practice if you want to make sure that a `switch` contains all enum values, because the compiler will only warn you if the `switch` has no `default` and the enum type is known.

> and as expected it caused compilation failures which I tried but am not skilled enough to fix.

Did you change the `enum` to `enum class`, or what was the change you were after?

Reply:
Thank you for explaining, it means a lot :)

> Did you change the enum to enum class, or what was the change you were after?

Yes, I used `enum class` and read about it in the docs, more out of curiosity.